xapian-applications/omega/postprocess.py

```diff
@@ -130,16 +127,18 @@ def generate_final_log(search_log, clicks_log, final_log):
                 did_to_count = qid_to_clicks[row[QUERYID]]
                 for index, did in enumerate(clicklist):
                     if did in did_to_count:
-                        clicklist[index] = did + ':' + str(did_to_count[did])
+                        clicklist[index] = '%s:%i' % (did, did_to_count[did])
                     else:
-                        clicklist[index] = did + ':' + '0'
-            else:
+                        clicklist[index] = did + ':0'
+            elif not clicklist:
```

Comment:
This will never run unless the hitlist is empty, which I don't think is what you want.